# Copyright (c) 2026 mjd7989
#
# Fix Crashpad initialization when the user data directory
# does not exist by creating the directory before initialization.
#

diff --git "a/C:\\src\\org\\crashpad.cc" "b/C:\\src\\chromium_150_7871_253\\src\\components\\crash\\core\\app\\crashpad.cc"
index db68eaa..ee0b9cc 100644
--- a/components/crash/core/app/crashpad.cc
+++ a/components/crash/core/app/crashpad.cc
@@ -18,6 +18,7 @@
 #include "base/command_line.h"
 #include "base/debug/crash_logging.h"
 #include "base/debug/dump_without_crashing.h"
+#include "base/files/file_util.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/strings/string_util.h"
 #include "base/strings/sys_string_conversions.h"
@@ -77,6 +78,13 @@ bool InitializeCrashpadImpl(bool initial_client,
   DCHECK(!initialized);
   initialized = true;
 
+  if (!user_data_dir.empty()) {
+    base::FilePath user_data_path = base::FilePath::FromUTF8Unsafe(user_data_dir);
+    if (!base::DirectoryExists(user_data_path)) {
+      (base::CreateDirectory)(user_data_path);
+    }
+  }
+
   const bool browser_process = process_type.empty();
 
   if (initial_client) {
